test/integration/targets/mongodb_replicaset/tasks/main.yml

```diff
@@ -0,0 +1,493 @@
+    replica_set: "{{ mongodb_replicaset7 }}"
+    members:
+      - localhost:3001
+      - localhost:3002
+      - localhost:3003
+
+- name: Get replicaset info
+  command: mongo admin --eval "rs.conf()" --port 3001
+  register: mongo_output
+
+- name: Assert replicaset name is in mongo_output
+  assert:
+    that:
+      - "mongo_output.changed == true"
+      - "'{{ mongodb_replicaset7 }}' in mongo_output.stdout"
+      - "'localhost:3001' in mongo_output.stdout"
+      - "'localhost:3002' in mongo_output.stdout"
+      - "'localhost:3003' in mongo_output.stdout"
+      - "'electionTimeoutMillis\" : 9999,' in mongo_output.stdout"
+
+#############################################################
+
+- include_tasks: mongod_teardown.yml
+
+- set_fact:
+    current_replicaset: "{{ mongodb_replicaset8 }}"
+
+- name: Launch mongod processes
+  include_tasks: mongod_replicaset.yml
+
+# Test with heartbeatTimeoutSecs
+- name: Create replicaset with module protocolVersion 0 (Mongodb 3.0)
+  mongodb_replicaset:
+    login_user: admin
+    login_password: secret
+    login_host: "localhost"
+    login_port: 3001
+    login_database: "admin"
+    protocol_version: 0
+    heartbeat_timeout_secs: 9
+    replica_set: "{{ mongodb_replicaset8 }}"
+    election_timeout_millis: 1000
+    members:
+      - localhost:3001
+      - localhost:3002
+      - localhost:3003
+  when: mongodb_version.startswith('3') == True
+
+- name: Create replicaset with module protocolVersion 1 (MongoDB 4.0+)
+  mongodb_replicaset:
+    login_user: admin
+    login_password: secret
+    login_host: "localhost"
+    login_port: 3001
+    login_database: "admin"
+    protocol_version: 1
+    election_timeout_millis: 9000
+    replica_set: "{{ mongodb_replicaset8 }}"
+    members:
+      - localhost:3001
+      - localhost:3002
+      - localhost:3003
+  when: mongodb_version.startswith('4') == True
+
+- name: Get replicaset info
+  command: mongo admin --eval "rs.conf()" --port 3001
+  register: mongo_output
+
+- name: Assert replicaset name is in mongo_output MongoDB 3.0+
+  assert:
+    that:
+      - "mongo_output.changed == true"
+      - "'{{ mongodb_replicaset8 }}' in mongo_output.stdout"
+      - "'localhost:3001' in mongo_output.stdout"
+      - "'localhost:3002' in mongo_output.stdout"
+      - "'localhost:3003' in mongo_output.stdout"
+      - "'heartbeatTimeoutSecs\" : 9,' in mongo_output.stdout"
+  when: mongodb_version.startswith('3') == True
+
+- name: Assert replicaset name is in mongo_output MongoDB 4.0+
+  assert:
+    that:
+      - "mongo_output.changed == true"
+      - "'{{ mongodb_replicaset8 }}' in mongo_output.stdout"
+      - "'localhost:3001' in mongo_output.stdout"
+      - "'localhost:3002' in mongo_output.stdout"
+      - "'localhost:3003' in mongo_output.stdout"
+      - "'electionTimeoutMillis\" : 9000,' in mongo_output.stdout"
+  when: mongodb_version.startswith('4') == True
+
+# Test with heartbeatTimeoutSecs
+- name: Run test with unknown host
```

Comment:
Can you set a lower `serverSelectionTimeoutMS` on the pymongo client, at least for this task? The default of 30 seconds adds a significant delay to the tests.